src/main/webapp/assets-lti/js/controllers/controllers.js

```diff
@@ -802,14 +801,15 @@ canvasSupportApp.controller('gradesController', ['$scope', '$location', '$rootSc
         Papa.parse(reader.result, {
         	complete: function(results) {
             $timeout(function(){
-              //when parsed extract the headers
-              $scope.headers = [results.data[0],results.data[1]];
-              // brittle - but all samples confirm it - the Points Possible cell is the 6th one in the second row
-              $scope.pointsPossible = $scope.headers[1][5];
-              // could also see if first cell is a "Points Possible" (with trimming) value
-              // and then cycle throght the remainder cells in that row till we get a value
-              //$log.info(valueToPluck);
-              $scope.pluckPos = _.indexOf($scope.headers[0], valueToPluck);
+              //remove any trailing cells and get name and pp as the last element
+              var assigName = _.last(_.compact(results.data[0]));
+              var pointsPossible = _.last(_.compact(results.data[1]));
+              //we are constructing the headers from scratch, since there is so much
+              //variability on the format, and all we are going to inherit
+              //from the input is the login id, points possible, and grades
+              $scope.headers = [['Student','ID','SIS User ID','SIS Login ID',assigName],['Points Possible','','','',pointsPossible]];
```

Comment:
What is this` '','',''` mean?

Reply:
The silly headers expected by Canvas have/should have this format.